solidstart-tanstackquery-tailwind-modules/src/components/RepoHeader/RepoHeader.tsx

```diff
@@ -0,0 +1,58 @@
+import { useLocation, useParams } from '@solidjs/router';
+import styles from './RepoHeader.module.css';
+import { createTabList } from './tabList';
+import { RepoHeading } from '../RepoHeading';
+import { RepoActionButtons } from '../RepoActionButtons';
+import { TabNavigation } from '../TabNavigation';
+import { splitProps } from 'solid-js';
+
+export type RepoHeaderProps = {
+  openIssueCount: number;
+  openPullRequestCount: number;
+  forkCount: number;
+  stargazerCount: number;
+  watcherCount: number;
+  visibility: string;
+  isOrg: boolean;
+};
+
+function RepoHeader(props: RepoHeaderProps) {
+  const [local] = splitProps(props, [
+    'openIssueCount',
+    'openPullRequestCount',
+    'forkCount',
+    'isOrg',
+    'stargazerCount',
+    'watcherCount',
+    'visibility',
+  ]);
+  const { pathname } = useLocation();
+  const params = useParams();
+
+  return (
+    <div class={styles.wrapper}>
+      <div class={styles.topRow}>
+        <RepoHeading visibility={local.visibility} isOrg={local.isOrg} />
+        <RepoActionButtons
+          forkCount={local.forkCount}
+          stargazerCount={local.stargazerCount}
+          watcherCount={local.watcherCount}
+        />
+      </div>
+      <div class={styles.bottomRow}>
+        <TabNavigation
+          //@ts-ignore
```

Comment:
after changing the type as described in the following comments, we can be able to remove this